travel-agency-service/src/main/java/cz/muni/fi/pa165/bluebat/facade/TripFacadeImpl.java

```diff
@@ -45,45 +46,46 @@ public TripFacadeImpl(TripService tripService,BeanMappingService beanMappingServ
     }
 
     @Override
-    public void createTrip(TripCreateDTO dto) {
+    public TripDTO createTrip(TripCreateDTO dto) {
+        Validator.NotNull(dto,"TripCreateDTO");
         Trip newTrip = new Trip();
         newTrip.setName(dto.getName());
         newTrip.setDestination(dto.getDestination());
         newTrip.setDateFrom(dto.getDateFrom());
         newTrip.setDateTo(dto.getDateTo());
         newTrip.setAvailableTrips(dto.getAvailableTrips());
         tripService.create(newTrip);
+        return beanMappingService.mapTo(newTrip,TripDTO.class);
 
     }
 
 
     @Override
-    public void updateTrip(TripDTO dto) {
+    public TripDTO updateTrip(TripDTO dto) {
+        Validator.NotNull(dto,"TripDTO");
         Trip update = tripService.findById(dto.getId());
-        if(update == null){
-            throw new IllegalArgumentException() {};
-        }
+        Validator.Found(update,"TripDTO");
```

Comment:
Zbytečná kontrola, jestli daný trip existuje, protože stejnou děláš i v TripService, proto bych to tady odebral.